spec/gears/string_spec.lua

```diff
@@ -51,11 +51,17 @@ describe("gears.string", function()
         assert.is_same(gstring.split("foo\n", "\n"), {"foo", ""})
         assert.is_same(gstring.split("foo\nbar", "\n"), {"foo", "bar"})
 
+        assert.is_same(gstring.split("", "%."), {""})
+        assert.is_same(gstring.split(".", "%."), {"", ""})
+        assert.is_same(gstring.split("foo", "%."), {"foo"})
+        assert.is_same(gstring.split("foo.", "%."), {"foo", ""})
+        assert.is_same(gstring.split("foo.bar", "%."), {"foo", "bar"})
+
         assert.is_same(gstring.split("", "."), {""})
-        assert.is_same(gstring.split(".", "."), {"", ""})
-        assert.is_same(gstring.split("foo", "."), {"foo"})
-        assert.is_same(gstring.split("foo.", "."), {"foo", ""})
-        assert.is_same(gstring.split("foo.bar", "."), {"foo", "bar"})
+        assert.is_same(gstring.split("a", "."), {"", ""})
+        assert.is_same(gstring.split("foo", "."), {"", "", "", ""})
```

Comment:
this is not normal - API compatibility shouldn't be broken within major release

Reply:
Okay, what about to add third parameter? The flag which will enable pattern matching instead of string.

Reply:
Or would it be better to create a separate function like `psplit`?

Reply:
depending on implementation - as far as i see from the diff implementation almost totally different, so i think 2 separate functions would be alright

Reply:
also as far as i see from codecov warnings - not all the code branches are covered by the new tests

Reply:
Done.